includes/db/DatabaseError.php

```diff
@@ -17,16 +17,22 @@ class DBError extends MWException {
 	 * @param $error String A simple error message to be used for debugging
 	 */
 	function __construct( DatabaseBase &$db, $error ) {
+		global $wgDBcluster;
+
 		$this->db = $db;
 		parent::__construct( $error );
 
+		$isMaster = !is_null( $db->getLBInfo( 'master' ) );
```

Comment:
Good point. Currently ``$isMaster`` variable is set to `true` when `$db` is the connection to the first machine on load balancer list.

We may want to modify [the LoadBalancer class](https://github.com/Wikia/app/blob/dev/includes/db/LoadBalancer.php#L614-614) and set the custom flag on `Database` when `$i == 0`.